services/external/google/user-client.ts

```diff
@@ -0,0 +1,162 @@
+import { logger } from "@/lib/logger";
+import * as Sentry from "@sentry/nextjs";
+import { GaxiosError } from "gaxios";
+import { google } from "googleapis";
+
+export function buildGoogleUserClient(token: GoogleUserCredential) {
+	const clientId = process.env.GOOGLE_OAUTH_CLIENT_ID;
+	if (!clientId) {
+		throw new Error("GOOGLE_OAUTH_CLIENT_ID is empty");
+	}
+	const clientSecret = process.env.GOOGLE_OAUTH_CLIENT_SECRET;
+	if (!clientSecret) {
+		throw new Error("GOOGLE_OAUTH_CLIENT_SECRET is empty");
+	}
+
+	const loggerWithSentry = {
+		error: (error: Error) => {
+			if (process.env.NODE_ENV === "development") {
+				console.error(error);
+			}
+			Sentry.captureException(error);
+		},
+		warning: (message: string) => {
+			if (process.env.NODE_ENV === "development") {
+				console.warn(message);
+			}
+			Sentry.captureMessage(message, "warning");
+		},
+		info: (message: string) => {
+			console.info(message);
+		},
+	};
+
+	return new GoogleUserClient(token, clientId, clientSecret, loggerWithSentry);
+}
+
+// MARK: Errors
+
+class GoogleTokenRefreshError extends Error {
+	constructor(message: string, options?: { cause?: unknown }) {
+		super(message, options);
+		this.name = this.constructor.name;
+		Object.setPrototypeOf(this, GoogleTokenRefreshError.prototype);
+	}
+}
+
+/**
+ * Determines if the given error requires authorization.
+ * if return value is true, the user should be redirected to the authorization page.
+ *
+ * see {@link https://supabase.com/docs/reference/javascript/auth-signinwithoauth}
+ *
+ * @param error - The error to check.
+ * @returns True if the error requires authorization, false otherwise.
+ */
+
+export function needsAuthorization(error: unknown) {
+	if (error instanceof GoogleTokenRefreshError) {
+		return true;
+	}
+	logger.debug({ error }, "error------------");
+	if (error instanceof GaxiosError) {
+		return error.status === 401 || error.status === 403 || error.status === 404;
+	}
+	return false;
+}
+
+export type GoogleUserData = {
+	name: string;
+	email: string;
+};
+
+class GoogleUserClient {
+	private clientId: string;
+	private clientSecret: string;
+	private logger: Logger;
```

Comment:
I've found that this `logger` is not used in this class.
If you don't need the reporting sentry feature here, consider remove this property.

Reply:
you're right.
I remove it by c0320c6